### ISSUE TITLE
"timestamp out of range" error in library search

### ISSUE THREAD

**Issue by gorobotgo** [pkkid/python-plexapi#675]
**Describe the issue**
library section search results in "OverflowError: timestamp out of range for platform time_t". 

**Code snipppets**
stack trace follows. My line of code that triggers the issue is on the fourth line (in "syncMovies"):

  File "plexSync.py", line 61, in <module>
    syncMovies (server_1_name, conn_1, server_2_name, conn_2, 'Movies', args.verbose, args.debug)
  File "/home/pi/scripts/plexSyncLib.py", line 58, in syncMovies
    **targetServer_movies = set(list(map((lambda x: x.title), targetConn.library.section(libraryName).search())))**
  File "/usr/local/lib/python3.7/dist-packages/plexapi/library.py", line 686, in search
    container_size=container_size, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/plexapi/library.py", line 381, in fetchItems
    items = self.findItems(data, cls, ekey, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/plexapi/base.py", line 227, in findItems
    item = self._buildItemOrNone(elem, cls, initpath)
  File "/usr/local/lib/python3.7/dist-packages/plexapi/base.py", line 98, in _buildItemOrNone
    return self._buildItem(elem, cls, initpath)
  File "/usr/local/lib/python3.7/dist-packages/plexapi/base.py", line 90, in _buildItem
    return ecls(self._server, elem, initpath)
  File "/usr/local/lib/python3.7/dist-packages/plexapi/base.py", line 53, in __init__
    self._loadData(data)
  File "/usr/local/lib/python3.7/dist-packages/plexapi/video.py", line 292, in _loadData
    Video._loadData(self, data)
  File "/usr/local/lib/python3.7/dist-packages/plexapi/video.py", line 44, in _loadData
    self.addedAt = utils.toDatetime(data.attrib.get('addedAt'))
  File "/usr/local/lib/python3.7/dist-packages/plexapi/utils.py", line 206, in toDatetime
    value = datetime.fromtimestamp(int(value))
OverflowError: timestamp out of range for platform time_t

**Expected behavior**
Should return the list of items from the library without an OverflowError

**Enviroment (please complete the following information):**
 - OS: execution is on Raspbian Buster, target Plex library is on Windows 10
 - Plex version 1.21.4.4079
 - Python Version 3.7.3
 - PlexAPI version 4.4.0

**Additional context**
This is only occurring for me against a Plex instance on Windows 10, works without issue on two other Plex libraries on macOS and Ubuntu.

**Comment by Hellowlol** [comment]
Are you using 32 bit python?

**Comment by gorobotgo** [comment]
> Are you using 32 bit python?

Yes.

**Comment by JonnyWong16** [comment]
There is a movie in your library with an `addedAt` date way in the past (`<-2^63` seconds) or way in the future (`>2^63` seconds). We can add better error handling for these edge cases but you should go fix that movie on your Plex server.

**Comment by gorobotgo** [comment]
> There is a movie in your library with an `addedAt` date way in the past (`<-2^63` seconds) or way in the future (`>2^63` seconds). We can add better error handling for these edge cases but you should go fix that movie on your Plex server.

Interesting, because all of my Plex servers are referencing media that is stored on the same NAS, and all of the movies on this server were "added" to the Plex library just a couple of days ago. I guess I'll try deleting the library and re-adding them? Or do you have a suggestion for identifying which movie(s) are the problem?

**Comment by Hellowlol** [comment]
Use a loop

**Comment by JonnyWong16** [comment]
Sort your library "By Date Added" in Plex Web.

**Comment by gorobotgo** [comment]
> Sort your library "By Date Added" in Plex Web.

Thanks, this allowed me to figure it out and fix it, though it highlighted some oddities in Plex. For one -- and I haven't figured out why this is -- things that I had added all at once were showing "Date Added" dates that spanned decades. Further, there is a disparity in how Plex displays date added in the Grid and List views; the "problem" media had a Date Added of nearly 50 years in the future, but in the Grid view, it showed as having been added "a few seconds ago". I "touch"ed the file, but Plex still showed the wrong Date Added, so I removed the folder from the library, scanned, emptied trash, re-added the folder, and re-scanned, and suddenly all of my movies show with the correct Date Added, including the one that had the impossible date. Now I can run my script without encountering the timestamp error.

Still probably something that PlexAPI should handle more gracefully, but this is a reasonable fix in the interim.
